nix/zlib-fdopen.patch

```diff
@@ -0,0 +1,23 @@
+diff --git a/deps/freetype/zlib/zutil.h b/deps/freetype/zlib/zutil.h
```

Comment:
is this patch really necessary? I *really* don't like the precedent of dumping a lot of stuff into wezterm's repo just for nix--I'm already on the fence about retaining the lock and nix files in here as it is.

Can you find a way to achieve whatever this is doing, without the patch?

Reply:
If we want to pin `zlib` `1.2.11`, yes.

The other option would be to bump `zlib` to `1.3.1`. I am not sure if this will bring other issues though, but at least this issue should be solved.

Reply:
1.3.1 sounds good to me; I've pushed that dep update.  Please rebase this PR and we'll see what happens in CI!

Reply:
BTW, if you can bump `libpng` we can also remove the other patch probably.

Reply:
just pushed a bump for libpng -> 1.6.44

Reply:
Done, thanks @wez.